### DISCUSSION TITLE
Alternate Parts/Help with shopping list

### DISCUSSION THREAD

**Discussion by Rol3oT** [makerforgetech/modular-biped#43] [Q&A] [answered]
I am following the project and trying to acquire parts. I am following [Electronics](https://github.com/dmt-labs/modular-biped/wiki/Electronics) wiki page to gather parts, but I am having troubles with following:

-9 SG9010 servos for the legs : I cannot find these on DigiKey or Adafruit, is it possible there is a typo? I have found MG92B servos no problem listed on Adafruit and DigiKey. The closest I got trough the Google Search is [this servo](https://www.pishop.ca/product/sg90-180-degrees-9g-micro-servo-motor-tower-pro/) but I am not sure. Did anyone use alternate servos that worked?

-An MPU6050 accelerometer : I haven't dived in to assembly instructions of this project just yet, but I am assuming we need an expansion board here like [this one](https://www.digikey.ca/en/products/detail/adafruit-industries-llc/3886/10709725)?

-USB-C PD(power source and switch) : Learned something new! First time seeing this module, but this [article](https://hackaday.com/2023/01/09/all-about-usb-c-power-delivery/) was very helpful. Looks like something like [this](https://www.amazon.ca/Docking-Station-MOKiN-Charging-ThinkPad/dp/B09YYBSN71/ref=sr_1_9?crid=20KYOKCL5MM5O&keywords=usb-c%2Bpd%2Bdock&qid=1692029544&s=electronics&sprefix=usb-c%2Bpd%2Bdock%2Celectronics%2C103&sr=1-9&th=1) should work? Any specifications to watch out for?

-Two buck converters (DC-DC) to step down the voltage to 5V for the Pi and for the arduino : I am lost here, could someone provide an example part and then I can find something similar? Shouldn't USB-C from USB-C PD step down to power up Raspberry Pi 4(lets say) anyway? 

I have just stumbled across this project couple of days ago and would really like to build one! Any help would be greatly appreciated and I hope it even helps someone else get answers on similar questions.

**Comment by danic85** [comment, marked as answer]
Hi, welcome!

Sure I can answer those:

- Yes that's a typo - it is actually SG5010, I have updated the wiki. If you have trouble sourcing them, any standard servo with a similar stall torque and voltage should work, but you may need to adjust the fit on the 3D print slightly. Here is an [example](https://www.adafruit.com/product/155).

- For the MPU6050 you may find more luck searching for GY-521 as that was the module I used, something like [this](https://www.az-delivery.uk/products/gy-521-6-achsen-gyroskop-und-beschleunigungssensor). It should only matter because that footprint is what the PCB was designed for, but as long as it's i2c it should work.

- For the USB-C PD module, I used a module like [this](https://www.ebay.co.uk/itm/294359311269?mkcid=16&mkevt=1&mkrid=711-127632-2357-0&ssspo=Yv_QNsB4RWO&sssrc=2047675&ssuid=&var=592976584536&widget_ver=artemis&media=COPY). It allows you to select the output voltage, anything above 5v should work. The goal is to pick a supply voltage higher than the 5v needed for the robot so that both buck converters have enough current to avoid any brownouts under heavy load. But someone may correct my logic there ;). I had problems with this early but the dual buck converter set up seems to fix those issues. I run mine on 8V.

- For the buck converter, I use the [XL4005](https://www.ebay.co.uk/itm/235084143758?mkcid=16&mkevt=1&mkrid=711-127632-2357-0&ssspo=h0eGMG00RM2&sssrc=2047675&ssuid=&var=535058051673&widget_ver=artemis&media=COPY) for the head and [XL4015](https://www.ebay.co.uk/itm/265241859904?mkcid=16&mkevt=1&mkrid=711-127632-2357-0&ssspo=YRTmJau9QrW&sssrc=2047675&ssuid=&widget_ver=artemis&media=COPY) for the body, but I expect the XL4005 should be able to cope in both places. Honestly anything with similar spec will do.

Bear in mind that these parts have been chosen because they are (relatively) inexpensive and readily available, at least where I am. If that isn't the case feel free to choose something similar. Worst case you'll need to adjust the CAD model a little to fit the new part.

I hope that helps, let us know if you need anything else clarifying. Great to see you on the community!

**Comment by Rol3oT** [comment]
I have also found 3D printed bearings and will try to see if they would be sufficient. It will save on cost if they work. 
https://www.printables.com/model/541192-print-in-place-bearings

**Comment by WhoIsMrSentry** [comment]
Hi wellcome discussion @Rol3oT  I am Emir, I am a 17 year old devloper I live in turkey and I am a contributor to this warehouse. I wanted to say if you haven't tried it, I made this robot using 3d printing bearings, spacers and screws, it both reduces the weight a bit and saves money. I used 608 bearing cable tie and 3d printing spacers I can send an image or video if you want details or I tried to use 3d printing completely in my own version which I made in a separate discussion

**Reply by WhoIsMrSentry** [reply]
it looks a bit in these photos, if you examine the directory, you can see my own versions of the companion robot that I made with my own comment, maybe it will be useful for you
https://github.com/SentryCoderDev/SentryBOT/blob/SentryBOT-V3/SentryBOT%20fotoğraflar/V4%20(5).jpg

https://github.com/SentryCoderDev/SentryBOT/blob/SentryBOT-V3/SentryBOT%20fotoğraflar/V4%20(4).jpg

**Reply by danic85** [reply]
@SentryCoderDev interesting! Do you have the files for the bearings etc? It would be good to share them if they work for you.

**Reply by WhoIsMrSentry** [reply]
@danic85 yes there is i will share all the files in a few hours

**Reply by WhoIsMrSentry** [reply]
@Rol3oT @danic85 Hi, sorry for the delay, I made a video with the files, I'm leaving the link below, you can click and watch the short video I explained how I did it I hope it works for you. Sorry in advance for my english it's not very good :/ 

Models:
[CompanionBOT_STL's.zip](https://github.com/dmt-labs/modular-biped/files/12379443/STL.s.zip)

Explain Video:
https://photos.app.goo.gl/9cSurLrkAbPcq93j9
